app/src/actions/vesselInfo.js

```diff
@@ -122,9 +121,59 @@ sub/seriesgroup=${seriesGroup}/${i}-01-01T00:00:00.000Z,${i + 1}-01-01T00:00:00.
   };
 }
 
+
+export function addVessel(seriesgroup, series = null, zoomToBounds = false) {
+  return (dispatch) => {
+    dispatch({
+      type: ADD_VESSEL,
+      payload: {
+        seriesgroup
+      }
+    });
+    dispatch(setCurrentVessel(seriesgroup));
+    dispatch(getVesselTrack(seriesgroup, series, zoomToBounds));
+  };
+}
+
 export function clearVesselInfo() {
   return {
-    type: CLEAR_VESSEL_INFO,
-    payload: false
+    type: CLEAR_VESSEL_INFO
+  };
+}
+
+export function toggleActiveVesselPin() {
+  return {
+    type: TOGGLE_ACTIVE_VESSEL_PIN
+  };
+}
+
+export function toggleVesselPin(seriesgroup) {
```

Comment:
I kind of see what you mean but here SRP + DRY: only the reducer knows about a given pinned track status. If we do a `pinVesselTrack/unpinVesselTrack` action, it means the view has knowledge of this status which is something I'd rather avoid. 

I'm open to suggestions.

TOGGLE_ACTIVE_VESSEL_PIN is a misleading action type though, will fix.